sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralExpressionSuite.scala

```diff
@@ -477,4 +477,299 @@ class LiteralExpressionSuite extends SparkFunSuite with ExpressionEvalHelper {
+            InternalRow(UTF8String.fromString("b"), 2)
+          )
+        ),
+        ArrayType(
+          StructType(Seq(StructField("col1", StringType), StructField("col2", IntegerType))),
+          containsNull = false
+        )
+      ),
+
+      // struct
+      Literal.create((1, 3.0, "abc")),
+      Literal(
+        InternalRow(true, 1.toLong),
+        StructType(
+          Seq(
+            StructField("col1", BooleanType, nullable = false).withComment("some-comment"),
+            StructField("col2", LongType)
+          )
+        )
+      ),
+
+      // struct contains array
+      Literal(
+        InternalRow(1.0, new GenericArrayData(Array(true, false, null))),
+        StructType(
+          Seq(
+            StructField("col1", DoubleType),
+            StructField("col2", ArrayType(BooleanType, containsNull = true))
+          )
+        )
+      ),
+
+      // map
+      Literal(
+        new ArrayBasedMapData(
+          new GenericArrayData(Array(UTF8String.fromString("a"), UTF8String.fromString("b"))),
+          new GenericArrayData(Array(1.toShort, 222.toShort))),
+        MapType(StringType, ShortType, valueContainsNull = false)
+      ),
+
+      // map contains array
+      Literal(
+        new ArrayBasedMapData(
+          new GenericArrayData(Array(UTF8String.fromString("a"), UTF8String.fromString("b"))),
+          new GenericArrayData(Array(
+            new GenericArrayData(Array(1, 2)), new GenericArrayData(Array(3))
+          ))
+        ),
+        MapType(StringType, ArrayType(IntegerType, containsNull = true))
+      )
+
+      // TODO(anchovyu): enable this test when map value comparison problem is fixed
+      // mixture of complex data type
+      // [map(
+      //   alex -> [(12345, home), (67890, work)],
+      //   bob -> [(null, home)]
+      // )]
+      //      , Literal(
+      //        new GenericArrayData(Array(
+      //          new ArrayBasedMapData(
+      //            new GenericArrayData(Array(
+      //              UTF8String.fromString("alex"),
+      //              UTF8String.fromString("bob"))
+      //            ),
+      //            new GenericArrayData(Array(
+      //              new GenericArrayData(Array(
+      //                InternalRow(12345, UTF8String.fromString("home")),
+      //                InternalRow(67890, UTF8String.fromString("work")))
+      //              ),
+      //              new GenericArrayData(Array(InternalRow(null, UTF8String.fromString("home"))))
+      //            ))
+      //          )
+      //        )),
+      //        ArrayType(
+      //          MapType(
+      //            StringType,
+      //            ArrayType(
+      //              StructType(Seq(
+      //                StructField("phone", IntegerType).withComment("phone number"),
+      //                StructField("tags", StringType)
+      //              ))
+      //            )
+      //          )
+      //        )
+      //      )
```

Comment:
Could we retain the line starts with `// TODO` and delete others?

Reply:
However the TODO itself is about enabling the commented out test below : )

Reply:
You can change it to
`TODO(anchovyu): fix the map value comparison problem`